presentation-compiler/src/main/dotty/tools/pc/printer/ShortenedNames.scala

```diff
@@ -220,7 +220,7 @@ object ShortenedNames:
     def apply(sym: Symbol)(using ctx: Context): ShortName =
       ShortName(sym.name, sym)
 
-  case class PrettyType(name: String) extends Type:
+  case class PrettyType(name: String) extends TermType:
```

Comment:
The fact that the PC defines its own subclass of Type is very weird, what is this used for?

Reply:
It's used for showing a shorter type without the prefix if that prefix is imported into scope. We will show more of the prefix if that is not imported. We could also figure out a way to fold that into the normal printer. We could have something like `ShortPrinter`

Reply:
As a first step, we could try to figure out what needs to change in Dotty to let the PC implement name shortening without hacks. PlainPrinter already defines `isOmittablePrefix` which could be overridden by the PC, but it needs to be generalized to also take the name of the thing we want to omit the prefix of as argument to support named imports. Once we get the PC to use this and remove `PrettyType` we can see what tests fail and proceed from there.